tapgarden/caretaker.go

```diff
@@ -1188,3 +1189,104 @@ func GenHeaderVerifier(ctx context.Context,
 		return err
 	}
 }
+
+// GenGroupVeifier generates a group key verification callback function given
+// a DB handle.
+func GenGroupVerifier(ctx context.Context,
+	mintingStore MintingStore) func(*btcec.PublicKey) error {
+
+	return func(groupKey *btcec.PublicKey) error {
+		if groupKey == nil {
+			return fmt.Errorf("cannot verify empty group key")
+		}
+
+		// This query will err if no stored group has a matching
+		// tweaked group key.
+		_, err := mintingStore.FetchGroupByGroupKey(
+			ctx, groupKey,
+		)
+		if err != nil {
+			assetGroupKey := groupKey.SerializeCompressed()
+			return fmt.Errorf("group %x not imported: %w",
+				assetGroupKey, err)
+		}
+
+		return nil
+	}
+}
+
+// GenGroupAnchorVerifier generates a caching group anchor verification callback
+// function given a DB handle.
+func GenGroupAnchorVerifier(ctx context.Context,
+	mintingStore MintingStore) func(*asset.Genesis,
+	*asset.GroupKey) error {
+
+	// Cache anchors for groups that were previously fetched.
+	groupAnchors := make(map[asset.SerializedKey]*asset.Genesis)
```

Comment:
`cache.LRU`? https://pkg.go.dev/github.com/lightninglabs/neutrino/cache@v1.1.2/lru#NewCache

Can also use above.

Reply:
Fixed size seems like a problem? We create these callbacks on startup and the size isn't bounded.

Reply:
Seems like something that could be added to `fn` with an Entry API similar to Rust HashMaps if we end up using this pattern a lot.